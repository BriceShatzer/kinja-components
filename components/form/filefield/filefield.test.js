import * as React from 'react';

import { shallow, mount } from 'enzyme';

import Filefield from './filefield';

const TEST_DESCRIPTION = 'Test';
const TEST_NAME = 'test';
const noop = () => {};

const stubElement = ({
	name = TEST_NAME,
	description = TEST_DESCRIPTION,
	onChange = noop,
	error = ''
} = {}) => (
	<Filefield
		onChange={onChange}
		name={name}
		description={description}
		error={error}
	/>
);

describe('<Filefield />', () => {
	it('should include name', () => {
		const wrapper = shallow(stubElement());
		expect(wrapper).toMatchSnapshot();
	});

	it('should trigger an onChange when file is selected', () => {
		const onChange = jest.fn();
		const wrapper = mount(stubElement({ onChange }));
		wrapper.find('input').simulate('change', {});
		expect(onChange).toHaveBeenCalled();
	});
});
