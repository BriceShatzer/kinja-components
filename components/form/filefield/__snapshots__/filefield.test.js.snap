// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filefield /> should include name 1`] = `
<Field
  error=""
  label="Test"
>
  <filefield__Input
    description="Test"
    error=""
    name="test"
    onChange={[Function]}
  />
</Field>
`;
